### ISSUE TITLE
Sparse hessian of vector dot product

### ISSUE THREAD

**Issue by thomas-rkk** [gdalle/sparsediffax#2]
Hi,

Thanks for providing the library.
I am in the process of wrapping my head around how it works and how to apply it.
I am solving an NLP with IPOPT and JAX, and am looking to provide sparse gradients and Hessians.

For that I also need to obtain the hessian of the dot product of the equality constraints and the Lagrange multipliers.
Would that be possible to do, in a similar manner to the current implementations?

This is my current dense implementation:

```
def hess_of_dot_prod(fun, x, v):
    def dotfun(x):
        return jnp.dot(fun(x), v)

    return jax.hessian(dotfun)(x)


con_eq_hessvp = lambda x, v: hess_of_dot_prod(eq_constraints, x, v)
```
Note: edited to avoid confusion with HVP's which are are different thing.
